packages/gatsby/src/query/query-runner.js

```diff
@@ -72,6 +72,7 @@ ${formatErrorDetails(errorDetails)}`)
     delete result.pageContext.pluginCreatorId
     delete result.pageContext.componentPath
     delete result.pageContext.context
+    delete result.pageContext.isCreatedByStatefulCreatePages
```

Comment:
@pvdz question here - should we try to set those to `undefined` instead of deleting it for better perf (given that we don't iterate over keys and just `JSON.stringify` it later on)?

Not part of this PR, because it applies to more stuff than what I'm adding, and don't want to tackle not related concerns within same PR.

Reply:
Yes. This goes to hidden object structures. A delete will cause some deopts to the object because the model that v8 thought an object had, is no longer valid. Setting it to undefined is a less intrusive operation (perf-wise, anyways). However, those keys will still show up in a json.
This is only really relevant for hot objects, that are constructed many times etc.